internal/app/connectconformance/results.go

```diff
@@ -413,13 +413,31 @@ func checkError(expected, actual *conformancev1.Error) multiErrors {
 	if len(actual.Details) < length {
 		length = len(actual.Details)
 	}
+	actualPayload := &conformancev1.ConformancePayload{}
+	expectedPayload := &conformancev1.ConformancePayload{}
 	for i := 0; i < length; i++ {
 		// TODO: Should this be more lenient? Are we okay with details getting re-ordered?
 		//       An alternative might be to create a map keyed by type, and for each type
 		//       remove expected messages as they are matched against actual ones.
-		if diff := cmp.Diff(expected.Details[i], actual.Details[i], protocmp.Transform()); diff != "" {
-			errs = append(errs, fmt.Errorf("actual error detail #%d does not match expected error detail: - wanted, + got\n%s",
-				i+1, diff))
+		actualDetails := actual.Details[i]
+		expectedDetails := expected.Details[i]
+
+		// If the error details is a conformance payload, then verify equality using the checkPayloads function
+		// Otherwise, just do a straight diff of the two
+		if actualDetails.MessageIs(actualPayload) && expectedDetails.MessageIs(expectedPayload) {
+			// TODO - Do these error conditions need checked? They should prob never fail because of the above check
```

Comment:
I think we can omit this TODO. We definitely want to check this, to make sure that the result we get from the client (possibly by way of the server) is correctly encoded.

But one thing we _do_ want to do is to skip the call to `checkPayloads` if either of the below `UnmarshalTo` fails. If unmarshaling failed, then we don't have a payload to check.